### ISSUE TITLE
[bug] Recipe upload skipped on a clean Artifactory repository

### ISSUE THREAD

**Issue by joaquin-herrero** [conan-io/conan#6186]
### Description
Trying to upload a package with all files to a new  artifactory repository skips the upload of the recipe. The binary packages are uploaded correctly. Artifactory shows the binary packages but no conan recipe.

### Environment Details:
  * Operating System+version: Debian 9
  * Compiler+version: 6.3
  * Conan version: 1.20.3
  * Conan revisions enabled
  * Python version: 3.5.2
  * Artifactory image: docker.bintray.io/jfrog/artifactory-cpp-ce:latest

### Steps to reproduce:
Create a new docker container from artifactory image and start it
```bash
$ docker pull docker.bintray.io/jfrog/artifactory-cpp-ce:latest
$ docker run --name artifactory-log -it --rm -v /home/joaquin/artifactory-logs/:/var/opt/jfrog/artifactory/logs/ -p 8081:8081 docker.bintray.io/jfrog/artifactory-cpp-ce:latest
```

Set up a new repository
```bash
$ conan remote add local http://localhost:8081/artifactory/api/conan/conan
```

Try to upload a recipe with all binary packages
```bash
$ conan upload -r local --all gtest/1.8.1@bincrafters/stable
```

### Logs

Package info:
```bash
$ conan info --paths gtest/1.8.1@bincrafters/stable
gtest/1.8.1@bincrafters/stable
    ID: 60d560fe9a3213f115a222bc64d6fdd92bd3741c
    BuildID: None
    export_folder: /home/joaquin/.conan/data/gtest/1.8.1/bincrafters/stable/export
    source_folder: /home/joaquin/.conan/data/gtest/1.8.1/bincrafters/stable/source
    build_folder: /home/joaquin/.conan/data/gtest/1.8.1/bincrafters/stable/build/60d560fe9a3213f115a222bc64d6fdd92bd3741c
    package_folder: /home/joaquin/.conan/data/gtest/1.8.1/bincrafters/stable/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c
    Remote: external=http://external:8081/artifactory/api/conan/external
    URL: http://github.com/bincrafters/conan-gtest
    Homepage: https://github.com/google/googletest
    License: BSD-3-Clause
    Author: Bincrafters <bincrafters@gmail.com>
    Topics: conan, gtest, testing, google-testing, unit-test
    Recipe: Cache
    Revision: 0
    Binary: Cache
    Binary remote: external
    Creation date: 2019-03-19 16:23:41
```

Error log:
```bash
$ conan upload -r local --all gtest/1.8.1@bincrafters/stable
Uploading to remote 'local':
Uploading gtest/1.8.1@bincrafters/stable to remote 'local'
Please log in to "local" to perform this action. Execute "conan user" command.
Remote 'local' username: admin
Please enter a password for "admin" account: 
Recipe is up to date, upload skipped
Uploading package 1/2: 73e2696d4559d3848507c61fd793c01fa2045cfb to 'local'
Compressing conan_package.tgz:  98%|#########7| 47/48 [00:00<00:00, 443.07files Compressing conan_package.tgz: 100%|##########| 48/48 [00:00<00:00, 445.84files /s]
Uploading conan_package.tgz: 100%|##########| 460k/460k [00:00<00:00, 28.3MB/s]
Uploading conaninfo.txt: 100%|##########| 532/532 [00:00<00:00, 172kB/s]
Uploading conanmanifest.txt: 100%|##########| 3.31k/3.31k [00:00<00:00, 1.81MB/s]
Uploading package 2/2: 60d560fe9a3213f115a222bc64d6fdd92bd3741c to 'local'
Uploading conan_package.tgz: 100%|##########| 1.20M/1.20M [00:00<00:00, 48.9MB/s]
Uploading conaninfo.txt: 100%|##########| 780/780 [00:00<00:00, 229kB/s]
Uploading conanmanifest.txt: 100%|##########| 4.79k/4.79k [00:00<00:00, 3.00MB/s]
```

Artifactory log:
```bash
2019-12-05 12:05:30,584 [http-nio-8081-exec-7] [INFO ] (o.a.r.HttpRepo      :470) - conan-remote downloading https://conan.bintray.com/v2/conans/gtest/1.8.1/bincrafters/stable/revisions/0/files/conanmanifest.txt 360 bytes 
2019-12-05 12:05:30,744 [http-nio-8081-exec-7] [INFO ] (o.a.r.HttpRepo      :483) - conan-remote downloaded  https://conan.bintray.com/v2/conans/gtest/1.8.1/bincrafters/stable/revisions/0/files/conanmanifest.txt 360 bytes at 2.29 KB/sec
2019-12-05 12:05:32,507 [http-nio-8081-exec-4] [INFO ] (o.a.r.HttpRepo      :470) - conan-remote downloading https://conan.bintray.com/v2/conans/gtest/1.8.1/bincrafters/stable/revisions/0/packages/73e2696d4559d3848507c61fd793c01fa2045cfb/revisions/0/files/conanmanifest.txt Unknown content length 
2019-12-05 12:05:32,537 [http-nio-8081-exec-4] [INFO ] (o.a.r.HttpRepo      :483) - conan-remote downloaded  https://conan.bintray.com/v2/conans/gtest/1.8.1/bincrafters/stable/revisions/0/packages/73e2696d4559d3848507c61fd793c01fa2045cfb/revisions/0/files/conanmanifest.txt 4.78 KB at 164.50 KB/sec
2019-12-05 12:05:32,587 [http-nio-8081-exec-6] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conan_package.tgz' Content-Length: 0
2019-12-05 12:05:32,602 [http-nio-8081-exec-8] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conan_package.tgz' Content-Length: 471441
2019-12-05 12:05:32,686 [art-exec-141] [INFO ] (o.a.a.c.s.ConanServiceImpl:278) - Starting Conan index calculation for conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb
2019-12-05 12:05:32,691 [http-nio-8081-exec-5] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conaninfo.txt' Content-Length: 0
2019-12-05 12:05:32,703 [http-nio-8081-exec-1] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conaninfo.txt' Content-Length: 532
2019-12-05 12:05:32,728 [art-exec-141] [INFO ] (o.a.a.c.s.ConanServiceImpl:280) - Conan index calculation for conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb finished successfully
2019-12-05 12:05:32,797 [http-nio-8081-exec-7] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conanmanifest.txt' Content-Length: 0
2019-12-05 12:05:32,808 [http-nio-8081-exec-2] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/73e2696d4559d3848507c61fd793c01fa2045cfb/0/conanmanifest.txt' Content-Length: 3385
2019-12-05 12:05:33,078 [http-nio-8081-exec-10] [ERROR] (o.a.a.c.r.ConanV2RemoteRepoHandler:433) - Unable to fetch file list from 'https://conan.bintray.com/v2/conans/gtest/1.8.1/bincrafters/stable/revisions/0/packages/60d560fe9a3213f115a222bc64d6fdd92bd3741c/revisions/f572ce9f3b87a4110ae47b8f59c9624c/files': HTTP/1.1 404 Not Found
2019-12-05 12:05:33,102 [http-nio-8081-exec-3] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conan_package.tgz' Content-Length: 0
2019-12-05 12:05:33,117 [http-nio-8081-exec-4] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conan_package.tgz' Content-Length: 1262739
2019-12-05 12:05:33,183 [art-exec-142] [INFO ] (o.a.a.c.s.ConanServiceImpl:278) - Starting Conan index calculation for conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c
2019-12-05 12:05:33,192 [http-nio-8081-exec-6] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conaninfo.txt' Content-Length: 0
2019-12-05 12:05:33,198 [art-exec-142] [INFO ] (o.a.a.c.s.ConanServiceImpl:280) - Conan index calculation for conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c finished successfully
2019-12-05 12:05:33,207 [http-nio-8081-exec-8] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conaninfo.txt' Content-Length: 780
2019-12-05 12:05:33,271 [http-nio-8081-exec-5] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conanmanifest.txt' Content-Length: 0
2019-12-05 12:05:33,279 [http-nio-8081-exec-1] [INFO ] (o.a.e.UploadServiceImpl:399) - Deploy to 'conan-local:bincrafters/gtest/1.8.1/stable/0/package/60d560fe9a3213f115a222bc64d6fdd92bd3741c/f572ce9f3b87a4110ae47b8f59c9624c/conanmanifest.txt' Content-Length: 4901
```

Artifactory image:
![Screenshot from 2019-12-05 13-59-50](https://user-images.githubusercontent.com/16034910/70237516-af662a00-1767-11ea-81b5-b4cfedaa4f18.png)

**Comment by joaquin-herrero** [comment]
It seems to fail on virtual repositories.

If I upload the package to conan-local it works as expected.

```bash
$ conan upload -r conan-local --all gtest/1.8.1@bincrafters/stable
Uploading to remote 'conan-local':
Uploading gtest/1.8.1@bincrafters/stable to remote 'conan-local'
Please log in to "conan-local" to perform this action. Execute "conan user" command.
Remote 'conan-local' username: admin
Please enter a password for "admin" account:
Uploading conan_export.tgz: 100%|##########| 765/765 [00:00<00:00, 413kB/s]
Uploading conan_sources.tgz: 100%|##########| 1.45k/1.45k [00:00<00:00, 1.08MB/s]
Uploading conanfile.py: 100%|##########| 4.04k/4.04k [00:00<00:00, 2.57MB/s]
Uploading conanmanifest.txt: 100%|##########| 360/360 [00:00<00:00, 155kB/s]
Uploaded conan recipe 'gtest/1.8.1@bincrafters/stable' to 'conan-local': http://localhost:8081/artifactory/api/conan/conan-local
Uploading package 1/2: 73e2696d4559d3848507c61fd793c01fa2045cfb to 'conan-local'
Package is up to date, upload skipped
Uploading package 2/2: 60d560fe9a3213f115a222bc64d6fdd92bd3741c to 'conan-local'
Package is up to date, upload skipped
```

**Comment by danimtb** [comment]
yes, this seems to be related only to virtual repositories.

Similar issues have been reported here https://www.jfrog.com/jira/browse/RTFACT-20712 when using Conan with virtual repositories in Artifactory.

Currently, it seems the behavior of virtuals is broken, so I'd recommend you to use just the conan-local repository.

Thanks for the report!

**Comment by memsharded** [comment]
Hi @joaquin-herrero 

It seems https://www.jfrog.com/jira/browse/RTFACT-20712 was solved in Artifactory 6.17. Did you manage to upgrade and test if your feature is gone?

**Comment by joaquin-herrero** [comment]
@memsharded yes, now it works as expected. Thank you!
